plugins/outputs/zabbix/README.md

```diff
@@ -0,0 +1,425 @@
+      }]}",
+      "clock":1519043805
+    }
+  ],
+  "clock":1519043805
+}
+```
+
+The Zabbix key is generated joining `lld`, the input name and tags (keys)
+alphabetically sorted.
+Some inputs could use different groups of tags for different fields, that is
+why the tags are added to the key, to allow having different discovery rules
+for the same input.
+
+The tags used in `value` are changed to uppercase to match the format of Zabbix.
+
+In the Zabbix server we should have a discovery rule associated with that key
+(telegraf.lld.net_response.port.protocol.server) and one item prototype for
+each field, in this case `result_type` and `response_time`.
+
+The item prototypes will be Zabbix trappers with keys (data type should also
+match and some values will be better stored as _delta_):
+
+```text
+telegraf.net_response.response_time[{#PORT},#PROTOCOL},{#SERVER}]
+telegraf.net_response.result_type[{#PORT},{#PROTOCOL},{#SERVER}]
+```
+
+The macros in the item prototypes keys should be alphabetically sorted so they
+can match the keys generated by this plugin.
+
+With that keys and the example trap, the host `myhost` will have two new items:
+
+```text
+telegraf.net_response.response_time[80,tcp,example.com]
+telegraf.net_response.result_type[80,tcp,example.com]
+```
+
+This plugin, for each metric, will send traps to the Zabbix server following
+the same structure (INPUT.FIELD[tags sorted]...), filling the items created by
+the discovery rule.
+
+In summary:
+
+- we need a discovery rule with the correct key and one item prototype for each
+field
+- this plugin will generate traps to create items based on the metrics seen in
+Telegraf
+- it will also send the traps to fill the new created items
+
+### Reducing the number of LLDs
+
+This plugin remembers which LLDs has been sent to Zabbix and avoid generating
+the same metrics again, to avoid the cost of LLD processing in Zabbix.
+
+It will only send LLD data each `lld_flush_count` intervals of this output
+plugin.
+
+For example, if the `flush_interval` for this plugins is set to `60s` and
+`lld_flush_count` set to 10, each 10 minutes Telegraf will send a metric to the
+discovery rules, but only if new data has been seen.
+
+But, could happen that package is lost or some host get new discovery rules, so
+each `lld_reset_count` the plugin will forget about the known data and start
+collecting again.
+
+For example, given `flush_interval=60s`, `lld_flush_count=10` and
+`lld_reset_count=6`:
+
+- the first 10 minutes this plugin will collect information about measurements
+and tags
+- in the 10th flush it will send the LLD data
+- in the next 10 min, it will keep collecting data
+- in the 20th flush, it will only send LLD data if new data has been seen
+- it will repeat this pattern till the 50th flush
+- from the 50th flush to the 60th flush it will collect new info, forgetting
+the already captured
+- the 60th flush will send LLD data seen between the 50th and the 60th flushes.
```

Comment:
There is no `lld_flush_count` or `lld_reset_count` configuration parameter..

Reply:
Fixed in cba0c2eaa